### PR TITLE
Update amp-autocomplete rich content example

### PR DESCRIPTION
**Pull request by caroqliu**
- Fixes the "Default state" example to display a category with each item on `focus` with no input
- Adds padding to the example inputs
- Resolves validation errors on `form` tags by including the proper attributes
- Removes input `type` attributes as [this change](https://github.com/ampproject/amphtml/pull/22470) allows
- Removes extraneous attributes where unnecessary so as to simplify the examples
- Displays the template code paired with the corresponding json data in the rich content templating section

### CODE REVIEW

File: examples/source/1.components/amp-autocomplete.html [hunk 1]
```diff
@@ -96,9 +100,11 @@
 
     A **datasource** must be a JSON object containing an array property `items`, and can be specified inline with a child `script type="application/json"` tag, or a server endpoint specified with a `src` attribute.
   -->
-  <form>
-    <amp-autocomplete filter="substring" min-characters="0">
-      <input type="text">
+  <form class="sample-form" method="post"
+        action-xhr="<% hosts.backend %>/components/amp-form/submit-form-input-text-xhr"
```
Comment (sebastianbenz) [inline examples/source/1.components/amp-autocomplete.html]: we're currently deprecating the ampbyexample backend. Can you please change all backend calls to:

`<% hosts.platform %>/documentation/examples/api/echo`

Comment (caroqliu) [reply examples/source/1.components/amp-autocomplete.html]: The change that resolves the error you are referring to isn't in prod yet, but should be fixed in the next push. Sorry if this change was a bit premature. 😇 

Making the other change now!

File: examples/source/1.components/amp-autocomplete.html [hunk 2]
```diff
@@ -152,17 +160,35 @@
...
+        }, {
+            "city" : "Trenton",
+            "state" : "New Jersey",
+            "areaCode" : 609,
+            "population" : 84964
+        }
+    ] }
+    ```
+
+    The corresponding display of these data in the amp-autocomplete can be specified through a template.
```
Comment (sebastianbenz) [inline examples/source/1.components/amp-autocomplete.html]: nit: 

```
s/amp-autocomplete/`amp-autocomplete`
```

**Comment by caroqliu** [comment]
@sebastianbenz Ready for merge :)

**Pull request merged**